nodejs/app.js

```diff
@@ -0,0 +1,68 @@
+const cookieParser = require("cookie-parser")
+const fetch = require("node-fetch")
+const express = require("express")
+const session = require("express-session")
+const OAuth2Strategy = require("passport-oauth2")
+const passport = require("passport")
```

Comment:
Can/should we use http://www.passportjs.org/docs/openid/ instead?

Reply:
The passport-openid library requires the user to enter their OpenID for authentication, not sure this is the expected user interaction like other examples? I also don't see anywhere it can set client_id/client_secret etc.

I tried to make this NodeJS implementation similar to Django and Flask examples. In those examples, it appeared that they were using the OAuth library, which I was also using OAuth in this NodeJS example.

I forgot the exact reasons why I ended up not using passport-openid, but I did spend a fair amount of time experimenting passport-openid and decided not to use it.